xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/DownloadAction.java

```diff
@@ -483,15 +488,43 @@ private boolean isValidRange(Long start, Long end)
         return start == null || end == null || end >= start;
     }
 
-    private boolean isAuthorized(String mimeType)
+    /**
+     * Check if an attachment should be downloaded or can be displayed inline.
+     * Attachments should be downloaded if:
+     * <ul>
+     *     <li>the request contains a force-download parameter with the value 1</li>
+     *     <li>or the mimetype is part of the forceDownload property list</li>
+     *     <li>or the attachments has not been added by a user with programming rights</li>
```

Comment:
The javadoc is not fully accurate. For example this line suggest that if the author of the attachment does not have PR it's always downloaded which is not really true.

Reply:
Indeed it's a bit ambiguous here.